### ISSUE TITLE
Use PSIS for fast cross-validation

### ISSUE THREAD

**Issue by ParadaCarleton** [alan-turing-institute/MLJ.jl#818]
I've built the [ParetoSmooth.jl](https://github.com/ParadaCarleton/ParetoSmooth.jl) package for fast approximate LOO cross-validation. The function currently only works with cross-entropy loss. (Bayesians usually call it expected log posterior density, and the package calls it log-score.) If you guys would like any help hooking it up to MLJ to speed up cross-validation while also improving its accuracy (by letting users take advantage of more folds), let me know. If anyone is interested in a pull request adding the ability to use leave-k-out instead of LOO, or adding loss functions other than cross-entropy, let me know.

**Comment by ablaom** [comment]
@ParadaCarleton Thanks for flagging this repository. 

What exactly is fast approximate LOO (leave-one-out) cross-validation? That is, how is it different from the usual LOOCV. I'm not aware of any method to speed up LOOCV unless one is restricting to some specific model class.

I looked up ParetoSmooth package but there is no docs and no readme. So probably hookup with the package is a bit premature?

Currently MLJ provides ordinary `CV `and `CVStratified` (with options to randomise the indices) and `CVTimeSeries`. A `repeats=...` option allows for Monte-Carlo CV.  One can use any metric you want and we currently provide about 60 metrics, including `LogLoss`. 

**Relevant MLJManual entries**:

[Estimating performance](https://alan-turing-institute.github.io/MLJ.jl/dev/evaluating_model_performance/)
[Performance measures](https://alan-turing-institute.github.io/MLJ.jl/dev/performance_measures/)
